Make ament_cmake a buildtool dependency

nwn opened the pull request:
The `rclcpp_action` package currently lists `ament_cmake` as a generic `<depend>` dependency, which implies that it's needed at both build-time and run-time. However, `ament_cmake` isn't used at runtime and so should only be listed as a `buildtool` dependency, as is done in most other packages.

nwn committed:
Make ament_cmake a buildtool dependency

The `ament_cmake` package isn't needed at runtime and so should only be
listed as a `buildtool_dependency`, as is done in most other packages.

Signed-off-by: Nathan Wiebe Neufeldt <nwiebeneufeldt@clearpath.ai>

clalancette commented on rclcpp_action/package.xml:
```diff
@@ -14,6 +14,7 @@
   <author email="dthomas@openrobotics.org">Dirk Thomas</author>
   <author email="jacob@openrobotics.org">Jacob Perron</author>
 
+  <buildtool_depend>ament_cmake</buildtool_depend>
```
Actually, we can just remove this completely.  This package already has a `buildtool_depend` on `ament_cmake_ros`, so this is redundant.

```suggestion
```

nwn replied:
Sounds good, removed.

nwn committed:
Remove the ament_cmake dependency entirely

Since there's already a buildtool dependency on ament_cmake_ros, having
ament_cmake as well is redundant.

Signed-off-by: Nathan Wiebe Neufeldt <nwiebeneufeldt@clearpath.ai>

clalancette commented:
Pulls: https://github.com/ros2/rclcpp/pull/2689
Gist: https://gist.githubusercontent.com/clalancette/f5185b026d33d01c407a865f8959b144/raw/886bab12bafaa4a0c567407e3b1a308101af9817/ros2.repos
BUILD args:  --packages-above-and-dependencies rclcpp_action
TEST args:  --packages-above rclcpp_action
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/14879
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=22191)](http://ci.ros2.org/job/ci_linux/22191/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=16443)](http://ci.ros2.org/job/ci_linux-aarch64/16443/)
* Linux-rhel [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-rhel&build=1712)](http://ci.ros2.org/job/ci_linux-rhel/1712/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=22941)](http://ci.ros2.org/job/ci_windows/22941/)
